persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -795,14 +808,26 @@ getColumn connectInfo getter tname [ PersistText cname
 
         cntrs <- liftIO $ with (stmtQuery stmt vars) (\src -> runConduit $ src .| CL.consume)
         pure $ case cntrs of
-            [] ->
-                Nothing
-            [[PersistText tab, PersistText ref, PersistInt64 pos, PersistText onDel, PersistText onUpd]] ->
+            [] -> Nothing
+            [[PersistText tab, schema, PersistText ref, PersistInt64 pos, PersistText onDel, PersistText onUpd]] ->
                 if pos == 1
-                then Just $ ColumnReference (EntityNameDB tab) (ConstraintNameDB ref) FieldCascade
-                    { fcOnUpdate = parseCascadeAction onUpd
-                    , fcOnDelete = parseCascadeAction onDel
-                    }
+                then Just $
+                  let colSchema =
+                        case schema of
+                          PersistNull -> Nothing
+                          PersistText schemaName -> do
+                            guard $ not (T.null schemaName)
+                            guard $ schemaName /= (pack $ MySQL.connectDatabase connectInfo)
+                            pure $ SchemaNameDB schemaName
```

Comment:
We got a schema name. If it's empty or the default, we return nothing, otherwise we return the name.